### ISSUE TITLE
Checkout does not stop loading

### ISSUE THREAD

**Issue by bve-9384** [evalentgroup/Magento-2-Ecster-Pay#2]
The Ecster Checkout page keeps loading forever in the following case:
- When at least one configurable product in the shopping cart has quantity = available stock quantity
- When only one shipping method is available or when another shipping method is selected

Increasing the stock quantity or decreasing the quantity in cart "solves" the issue. The problem only occurs with configurable products (variations).

Running plugin version 1.0.0 in test mode, verified on both Magento 2.3.4 and 2.3.5
Can see status code 400 Bad Request when V1/carts/mine/shipping-information is called 
"Out-of-Stock Threshold" is set to 0 on all levels and checkout works fine with Bank Transfer Payment instead of Ecster

**Comment by TobiasCodeNull** [comment]
Hi @bve-9384 

I was able to reproduce and fix the issue.
The issue seemed to be in ShippingInformationManagement that is overridden in the module.
The saving of the quote causes an listener (after save) to trigger which caused the qty to be added to the quote twice and thus giving an error when you were trying to buy the last items in stock. 

I've removed this from the code as the cart key is not interesting for the customer and it updates after each call to Ecster. There is another key which is used as reference to Ecsters order. This is called ecster_internal_reference on the order in the database.

Feel free to test by require the module as "dev-develop as 1.0.0". If it works on your end as well we are ready to release the fix.
